File: examples/javascript/search-leads.js
const apiKey = process.env.LASTDATABASE_API_KEY;

if (!apiKey) {
    console.error(
        "ERROR: LASTDATABASE_API_KEY is not set."
    );

    process.exitCode = 1;
} else {
    const url = new URL(
        "https://lastdatabase.com/api/leads/search"
    );

    url.searchParams.set(
        "type",
        "email"
    );

    url.searchParams.set(
        "country",
        "usa"
    );

    url.searchParams.set(
        "industry",
        "healthcare"
    );

    url.searchParams.set(
        "limit",
        "10"
    );

    async function searchLeads() {
        const response = await fetch(
            url,
            {
                headers: {
                    Authorization:
                        `Bearer ${apiKey}`,
                    Accept:
                        "application/json",
                },
            }
        );

        const body =
            await response.text();

        if (!response.ok) {
            throw new Error(
                `HTTP ${response.status}: ${body}`
            );
        }

        const data =
            JSON.parse(body);

        console.log(
            JSON.stringify(
                data,
                null,
                2
            )
        );
    }

    searchLeads().catch(
        (error) => {
            console.error(
                error.message
            );

            process.exitCode = 1;
        }
    );
}

File: examples/python/search_leads.py
import json
import os
import urllib.parse
import urllib.request

api_key = os.environ.get("LASTDATABASE_API_KEY")

if not api_key:
    raise SystemExit(
        "ERROR: LASTDATABASE_API_KEY is not set."
    )

params = urllib.parse.urlencode(
    {
        "type": "email",
        "country": "usa",
        "industry": "healthcare",
        "limit": 10,
    }
)

url = (
    "https://lastdatabase.com/api/leads/search?"
    + params
)

request = urllib.request.Request(
    url,
    headers={
        "Authorization": f"Bearer {api_key}",
        "Accept": "application/json",
    },
)

try:
    with urllib.request.urlopen(
        request,
        timeout=30,
    ) as response:
        payload = json.load(response)

except Exception as error:
    raise SystemExit(
        f"Request failed: {error}"
    )

print(
    json.dumps(
        payload,
        indent=2,
    )
)
